tests/integration/test_interface.py

```diff
@@ -45,7 +46,7 @@
 )
 from tests.unit.test_configuration import COLUMN_NOISE_TYPES
 
-DATASET_GENERATION_FUNCS: dict[str, Callable] = {
+DATASET_GENERATION_FUNCS: dict[str, Callable[..., Any]] = {
```

Comment:
Same here.

Reply:
All the Callables have the same arguments except one (generate_social_security) which is missing the state argument. Is there a clean way to specify this?

So most Callables would be 

Callable[[arg_1, arg_2, arg_3, arg_4, str | None, arg_5, arg_6]

except generate_social_security, which would be

Callable[[arg_1, arg_2, arg_3, arg_4, arg_5, arg_6]